examples/llm/inference/benchmarking/benchmark.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2022 MosaicML Examples authors
+# SPDX-License-Identifier: Apache-2.0
+
+import contextlib
+import sys
+import time
+
+import deepspeed
+import numpy as np
+import torch
+# You can use this to load the model weights
+from composer.core import get_precision_context
+from omegaconf import OmegaConf as om
+
+from examples.llm.src import COMPOSER_MODEL_REGISTRY
+
+
+def compare_precision(precision, param_dtype):
+    if ((precision == 'amp_bf16' and param_dtype != torch.bfloat16) or
+        (precision == 'amp_fp16' and param_dtype != torch.float16) or
+        (precision == 'fp32' and param_dtype != torch.float32)):
+        raise ValueError(
+            f'Precision type is: {precision} but model dtype is: {param_dtype}. '
+            f"The expected precision and model precision don't match.")
+
+
+def main(config):
+    inference_config = {
+        'replace_with_kernel_inject': True,
+        'dtype': torch.bfloat16,
+        'replace_method': 'auto',
+        'enable_cuda_graph': False,
+        'tensor_parallel': {
+            'tp_size': 0
+        },
+    }
+
+    model = COMPOSER_MODEL_REGISTRY[config.model.name](config.model,
+                                                       config.tokenizer)
+    model.eval()
+
+    tokenizer = model.tokenizer
+
+    if config.use_deepspeed:
+        model = deepspeed.init_inference(model, config=inference_config)
+
+        # Checking if deepspeed casts dtypes correctly
+        for _, p in model.named_parameters():
+            compare_precision(config.precision, p.dtype)
+            break
+    else:
+        model.to(torch.cuda.current_device())
+
+    n_params = sum(p.numel() for p in model.parameters())
+    print('n_params is: ', n_params)
+
+    print('Run name, latency (s), tokens per second')
+    print('=' * 75)
+
+    stats = []
+    for batch_size in config.batch_sizes:
+        for input_length in config.input_lengths:
+            for output_length in config.output_lengths:
+                times = []
+
+                batch = torch.randint(
+                    0,
+                    config.model.vocab_size - 1,
+                    size=(
+                        batch_size,
+                        input_length)).to(f'cuda:{torch.cuda.current_device()}')
+
+                # Make sure we are not generating a fake batch with a EOS token
+                eos_mask = batch == tokenizer.eos_token_id
+                batch = batch.masked_fill(eos_mask,
+                                          config.tokenizer.non_eos_token_id)
+
+                batch = batch.to(torch.long)
+
+                attention_mask = torch.logical_not(
+                    torch.eq(batch, tokenizer.eos_token_id))
```

Comment:
I'm not sure if this section is doing the right thing. We only want to avoid padding tokens, not EOS tokens. We can also remove the EOS token id in the generate_kwargs and that will force it to generate the `max_len` tokens every time.